### DISCUSSION TITLE
Variants with post date and available for purchase

### DISCUSSION THREAD

**Discussion by davidhellmann** [craftcms/commerce#2314] [Ideas]
Maybe more a Feature for the upcomming commerce Version but here is my Idea.

Is it possible to have Variants with a `postDate` ?
We import our products automatically but some of the variants have a date in the future (new Stuff). Any chance that craft can handle that? Disable until the date is reached?

Thanks! :)

**Comment by lukeholder** [comment]
We added a new checkbox to products in beta.5 which allows you to uncheck 'available for purchase', it isn't date based but would that be enough to satisfy the need to publish a product live without it being purchasable yet?

**Comment by davidhellmann** [comment]
Sounds interesting. I‘ll check that! Thanks :)

**Comment by davidhellmann** [comment]
Hm ah I see. But we need it more on variants level as on product level.

**Comment by lukeholder** [comment]
Thanks for the feedback.

**Comment by AugustMiller** [comment]
I bet this could be done with a custom field… it may require listening to the `BEFORE_SAVE_LINE_ITEM` event in a plugin/module to ensure someone sneaky hasn't added a prohibited purchasable to their cart, but otherwise, you could filter out variants from display based on that field, alone:

```twig
{# In a Product's template... #}
{% set timeSensitiveVariants = craft.variants({
  productId: product.id,
  myReadyDateField: ">= #{now}"
}).all %}
```

I don't think this is a _bad_ idea for core, but wanted to offer a means by which you could implement, today!

**Comment by lukeholder** [comment]
We added the following event to allow you to check the current user permissions or order to know if the variant should be available:

https://github.com/craftcms/commerce/pull/2109

Leaving this ticket open as we want to add  the `available` checkbox to the variants themselves.
